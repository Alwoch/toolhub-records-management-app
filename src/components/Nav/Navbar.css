nav{
 display:flex;
 height: 70px;
 align-items: center;
 justify-content: space-between;
 background-color: var(--primary-700);
 padding:1rem 2rem;
 color:var(--white);
 min-width: 300px;
}

#nav-list{
display: flex;
align-items: center;
justify-content: center;
}

#nav-list li{
 padding:0 1.5rem;
 position:relative;
}

#nav-list li a{
 color:var(--white);
 font-weight: 500;
 font-size: 1.3rem;
 transition:0.3s ease-in-out
}

#nav-list li a:hover,
#nav-list li a.active{
 color:var(--primary-200);
}

#nav-list li a:hover::after,
#nav-list li a.active::after{
 content:"";
 width:30%;
 height: 2px;
 background:var(--primary-200);
 position:absolute;
 bottom:-4px;
 left:25px;
}

#mobile{
 display:none;
}

@media only screen and (max-width:700px){
 #nav-list{
  display: flex;
  flex-direction: column;
  align-items:flex-start;
  justify-content: flex-start;
  position:fixed;
  top:70px;
  right:-300px;
  height:100vh;
  background-color:var(--black);
  transition: 0.3s ease-in-out;
  }

  #nav-list.active{
   right:0;
  }

  #mobile{
   display:block;
  }

  #nav-list li{
  margin-top: 1rem;
  }

  #mobile i{
   cursor:pointer;
   font-size:1rem;
  }
}