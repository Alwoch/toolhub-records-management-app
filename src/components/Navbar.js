const Navbar = () => {
  return (
    <nav>
      <button className="button button-hipster">button</button>
    </nav>
  );
};

export default Navbar;
