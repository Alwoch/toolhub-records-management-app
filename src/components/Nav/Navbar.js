import { useState } from "react";
import { Link } from "react-router-dom";
import "./Navbar.css";

const Navbar = () => {
  const [state, setState] = useState(false);

  const handleState = () => {
    setState(!state);
  };

  return (
    <nav className=".container-fluid" data-testid="nav">
      <h3>Toolhub</h3>
      <div>
        <ul id="nav-list" className={state ? "#nav-list active" : "#nav-list"}>
          <li>
            <Link className="active" to="/">
              Home
            </Link>
          </li>
          <li>
            <Link to="/dashboard">Dashboard</Link>
          </li>
          <li>
            <Link to="/leaderboard">Leaderboard</Link>
          </li>
        </ul>
      </div>
      <div id="mobile" onClick={handleState}>
        <i id="bar" aria-hidden="false" className={state ? "fas fa-times" : "fas fa-bars"}></i>
      </div>
    </nav>
  );
};

export default Navbar;
