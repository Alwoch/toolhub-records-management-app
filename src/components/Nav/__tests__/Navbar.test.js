import { render,screen} from '@testing-library/react';
import renderer from "react-test-renderer"
import { BrowserRouter } from "react-router-dom";
import Navbar from '../Navbar';

//Tests cases
  //Navbar should render correctly
  //Navbar should match the snapshot

test("should render Navbar component successfully",()=>{
 render(<BrowserRouter><Navbar/></BrowserRouter>);
 const NavComponent=screen.getByTestId('nav');
 expect(NavComponent).toBeInTheDocument();
})

test("Navbar should match snapshot",()=>{
 const navTree=renderer.create(<BrowserRouter><Navbar/></BrowserRouter>).toJSON()
 expect(navTree).toMatchSnapshot()
})