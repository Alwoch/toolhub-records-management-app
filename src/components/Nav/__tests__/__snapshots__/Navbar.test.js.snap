// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should match snapshot 1`] = `
<nav
  className=".container-fluid"
  data-testid="nav"
>
  <h3>
    Toolhub Records
  </h3>
  <div>
    <ul
      className="#nav-list"
      id="nav-list"
    >
      <li>
        <a
          className="active"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/dashboard"
          onClick={[Function]}
        >
          Dashboard
        </a>
      </li>
      <li>
        <a
          href="/leaderboard"
          onClick={[Function]}
        >
          Leaderboard
        </a>
      </li>
    </ul>
  </div>
  <div
    id="mobile"
    onClick={[Function]}
  >
    <i
      className="fas fa-bars"
      id="bar"
    />
  </div>
</nav>
`;
